lib/router/mbaas.js

```diff
@@ -64,55 +64,77 @@ function initRouter(mediator, authResponseExclusionList, expressSessionMiddlewar
+      res.json(result);
+      mediator.publish('done:wfm:user:read', result);
     });
-    mediator.publish('wfm:user:read', userId);
   });
+
   router.route('/:id').put(function(req, res) {
     var userId = req.params.id;
     var user = req.body.user;
-    mediator.once('done:wfm:user:update:' + userId, function(saveduser) {
-      res.json(saveduser);
+    mediator.request('wfm:user:update', user, {uid: userId}).then(function(result) {
+      if (result instanceof Error) {
+        mediator.publish('error:wfm:user:update', user);
+        return res.status(500).json({message: "Unexpected error when updating a user. " + result.toString()});
+      }
+      res.json(result);
+      mediator.publish('done:wfm:user:update', user);
     });
-    mediator.publish('wfm:user:update', user);
   });
+
   router.route('/').post(function(req, res) {
-    var ts = new Date().getTime();  // TODO: replace this with a proper uniqe (eg. a cuid)
     var user = req.body.user;
-    user.createdTs = ts;
-    mediator.once('done:wfm:user:create:' + ts, function(createduser) {
-      res.json(createduser);
+    mediator.request('wfm:user:create', user).then(function(result) {
+      if (result instanceof Error) {
+        mediator.publish('error:wfm:user:create', user);
+        return res.status(500).json({message: "Unexpected error when creating a user. " + result.toString()});
+      }
+      res.json(result);
+      mediator.publish('done:wfm:user:create', user);
     });
-    mediator.publish('wfm:user:create', user);
   });
+
   router.route('/:id').delete(function(req, res) {
     var userId = req.params.id;
     var user = req.body.user;
-    mediator.once('done:wfm:user:delete:' + userId, function(deleted) {
-      res.json(deleted);
+    mediator.request('wfm:user:delete', userId).then(function(result) {
+      if (result instanceof Error) {
+        mediator.publish('error:wfm:user:delete', user);
+        return res.status(500).json({message: "Unexpected error when deleting a user. " + result.toString()});
+      }
+      res.json(result);
+      mediator.publish('done:wfm:user:delete', user);
     });
-    mediator.publish('wfm:user:delete', user);
   });
   return router;
 }
 
 /**
 * Function to trim the authentication response to remove certain fields from being sent.
 * By default, the password will be removed from the response.
-* @param authResponse {object} - the untrimmed auth response
-* @param exclusionList {array} - the array of field names to remove from the authentication response
-* @return authResponse {object} - the trimmed authentication response
+* @param {object} authResponse - the untrimmed auth response
+* @param {array} exclusionList - the array of field names to remove from the authentication response
+* @return {object} authResponse - the trimmed authentication response
 */
 function trimAuthResponse(authResponse, exclusionList) {
-  if (exclusionList === undefined || exclusionList === null) {
+  if (!exclusionList) {
```

Comment:
`[]` is truthy, so it would also fall into this check. Maybe use [`_.isEmpty`](https://lodash.com/docs/4.17.4#isEmpty)